### PR TITLE
Bump bootstrap compiler

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
This bumps the bootstrap compiler and the rustfmt that x.py fmt uses.

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
Hm, CI failure is concerning. This seems to be due to the Cargo update (a bump to the Cargo.lock resolved it). @ehuss -- do we know what might be causing this? This seems like a bug, since this PR doesn't do anything other than update Cargo, so we shouldn't be seeing a Cargo.lock update as a result. Maybe there's a resolver change in beta that is leading to a forced update?

The "Bump Cargo.lock?" commit is the one I'm referencing.

**Comment by Mark-Simulacrum** [comment]
I'm going to r? @ehuss here for now as I'd like their input on the Cargo.lock bump being a bug before we merge, but if you don't want to review this feel free to re-r? after that to someone else.

**Comment by Mark-Simulacrum** [comment]
Huh. Weird. We don't seem to be passing --edition=2018 and I don't think we've made try a reserved keyword in 2015?

cc @rust-lang/release -- anyone know of changes relating to try! on beta?

File: src/libcore/macros/mod.rs [hunk 1]
```diff
@@ -360,7 +336,7 @@ macro_rules! r#try {
         }
     };
     ($expr:expr,) => {
-        $crate::r#try!($expr)
+        $crate::try!($expr)
```
Comment (jonas-schievink) [inline src/libcore/macros/mod.rs:339]: Was this change intentional?

**Comment by ehuss** [comment]
Your first commit modified `Cargo.lock` (https://github.com/rust-lang/rust/pull/71439/commits/dc11e4b9381cbb8339e529a6cced4fd8e48527ba), reverting a change made in #71402.  Maybe you updated your local tree without updating your submodules?  Not sure...

File: src/libcore/macros/mod.rs (hunk 1 above)
Comment (Mark-Simulacrum) [reply src/libcore/macros/mod.rs:339]: Oh. I suspect rustfmt? I'll look...

**Comment by Mark-Simulacrum** [comment]
I thought I had checked @ehuss -- but you are correct, I think. Sorry for the noise!

I believe I've pushed up a fix to both bugs.

r? @jonas-schievink since you've presumably done at least a little looking at the patch...

**Comment by Mark-Simulacrum** [comment]
@bors r=jonas-schievink

(rustfmt "bug" was my editor not passing --edition=2018).

File: src/librustc_metadata/rmeta/table.rs [hunk 2]
```diff
@@ -61,7 +58,7 @@ macro_rules! fixed_size_encoding_byte_len_and_defaults {
             };
             self.write_to_bytes(&mut b[i]);
         }
-    }
+    };
```
Comment (RalfJung) [inline src/librustc_metadata/rmeta/table.rs:61]: Curious spurious (but harmless) changes?

Comment (Mark-Simulacrum) [reply src/librustc_metadata/rmeta/table.rs:61]: These are rustfmt-caused (from the update, I'm pretty sure, not my editor :)

**Comment by Mark-Simulacrum** [comment]
CI failure is I think related to github troubles, not this PR's changes.

**Comment by jonas-schievink** [comment]
@bors p=1

(fairly prone to merge conflicts due to the spread-out changes)
